client/asset/zec/zec.go

```diff
@@ -92,238 +114,2323 @@ var (
 			IsBoolean: true,
 		},
 	}
-	// WalletInfo defines some general information about a Zcash wallet.
-	WalletInfo = &asset.WalletInfo{
-		Name:              "Zcash",
-		Version:           version,
-		SupportedVersions: []uint32{version},
-		UnitInfo:          dexzec.UnitInfo,
-		AvailableWallets: []*asset.WalletDefinition{{
-			Type:              walletTypeRPC,
-			Tab:               "External",
-			Description:       "Connect to zcashd",
-			DefaultConfigPath: dexbtc.SystemConfigPath("zcash"),
-			ConfigOpts:        configOpts,
-			NoAuth:            true,
-		}},
+	// WalletInfo defines some general information about a Zcash wallet.
+	WalletInfo = &asset.WalletInfo{
+		Name:              "Zcash",
+		Version:           version,
+		SupportedVersions: []uint32{version},
+		UnitInfo:          dexzec.UnitInfo,
+		AvailableWallets: []*asset.WalletDefinition{{
+			Type:              walletTypeRPC,
+			Tab:               "External",
+			Description:       "Connect to zcashd",
+			DefaultConfigPath: dexbtc.SystemConfigPath("zcash"),
+			ConfigOpts:        configOpts,
+			NoAuth:            true,
+		}},
+	}
+)
+
+func init() {
+	asset.Register(BipID, &Driver{})
+}
+
+// Driver implements asset.Driver.
+type Driver struct{}
+
+// Open creates the ZEC exchange wallet. Start the wallet with its Run method.
+func (d *Driver) Open(cfg *asset.WalletConfig, logger dex.Logger, network dex.Network) (asset.Wallet, error) {
+	return NewWallet(cfg, logger, network)
+}
+
+// DecodeCoinID creates a human-readable representation of a coin ID for
+// Zcash.
+func (d *Driver) DecodeCoinID(coinID []byte) (string, error) {
+	// Zcash shielded transactions don't have transparent outputs, so the coinID
+	// will just be the tx hash.
+	if len(coinID) == chainhash.HashSize {
+		var txHash chainhash.Hash
+		copy(txHash[:], coinID)
+		return txHash.String(), nil
+	}
+	// For transparent transactions, Zcash and Bitcoin have the same tx hash
+	// and output format.
+	return (&btc.Driver{}).DecodeCoinID(coinID)
+}
+
+// Info returns basic information about the wallet and asset.
+func (d *Driver) Info() *asset.WalletInfo {
+	return WalletInfo
+}
+
+// WalletConfig are wallet-level configuration settings.
+type WalletConfig struct {
+	UseSplitTx       bool   `ini:"txsplit"`
```

Comment:
**Shielded splits** are default behavior now when transparent funds are insufficient, but I left the purely transparent split machinery in place to avoid overlocks.